docs/manual/contributing.rst

```diff
@@ -48,7 +55,35 @@ to our `issue tracker <https://github.com/qtile/qtile/issues>`_ on GitHub.
 
     * **Code** that conforms to PEP8.
     * **Unit tests** that pass locally and in our CI environment (More below).
+      *Please add unit tests* to ensure that your code works and stays working!
     * **Documentation** updates on an as needed basis.
+    * A ``qtile migrate`` **migration** is required for config-breaking changes.
+      See `migrate.py <https://github.com/qtile/qtile/blob/libqtile/scripts/migrate.py>`_
+      for examples and consult the `bowler documentation <https://pybowler.io>`_
+      for detailed help and documentation.
+    * **Code** that does not include *unrelated changes*. Examples for this are
+      formatting changes, replacing quotes or whitespace in other parts of the
+      code or "fixing" linter warnings popping up in your editor on existing
+      code. *Do not include anything like the above!*
```

Comment:
Yeah I thought about that but wasn't quite sure if it would be too obvious. Also, this might encourage people to create pull requests with all sorts of linter fixes. It was decided that we don't want to pollute the git history. So we might want to leave that out for now and then put it in once the style guard (#2328) is doing its job.

Reply:
Fair enough :)